File: Shortest Distance to Target String in a Circular Array/Shortest Distance to Target String in a Circular Array.py
# Solution for Shortest Distance to Target String in a Circular Array in PY

class Solution:
    def closestTarget(self, words, target, startIndex):
        
        total_words = len(words)
        minimum_steps = float('inf')

        # Traverse all positions to find occurrences of target
        for current_index in range(total_words):

            # Check if current word matches target
            if words[current_index] == target:

                # Linear distance (direct movement)
                linear_distance = abs(current_index - startIndex)

                # Circular distance (wrap-around movement)
                circular_distance = total_words - linear_distance

                # Choose the minimum of both paths
                best_distance = min(linear_distance, circular_distance)

                # Update global minimum
                minimum_steps = min(minimum_steps, best_distance)

        # If target was never found, return -1
        return -1 if minimum_steps == float('inf') else minimum_steps

File: Shortest Distance to Target String in a Circular Array/Shortest Distance to Target String in a Circular Array.java
// Solution for Shortest Distance to Target String in a Circular Array in JAVA

class Solution {
    public int closestTarget(String[] words, String target, int startIndex) {
        
        int totalWords = words.length;
        int minimumSteps = Integer.MAX_VALUE;

        // Traverse all positions to find occurrences of target
        for (int currentIndex = 0; currentIndex < totalWords; currentIndex++) {

            // Check if current word matches target
            if (words[currentIndex].equals(target)) {

                // Linear distance (direct movement)
                int linearDistance = Math.abs(currentIndex - startIndex);

                // Circular distance (wrap-around movement)
                int circularDistance = totalWords - linearDistance;

                // Choose the minimum of both paths
                int bestDistance = Math.min(linearDistance, circularDistance);

                // Update global minimum
                minimumSteps = Math.min(minimumSteps, bestDistance);
            }
        }

        // If target was never found, return -1
        return (minimumSteps == Integer.MAX_VALUE) ? -1 : minimumSteps;
    }
}
